pkg/middleware/request_logger.go

```diff
@@ -0,0 +1,53 @@
+package middleware
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/justinas/alice"
+	middlewareapi "github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/middleware"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/logger"
+)
+
+func NewRequestLogger() alice.Constructor {
+	return requestLogger
+}
+
+func requestLogger(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		startTime := time.Now()
+		url := *req.URL
+
+		next.ServeHTTP(rw, req)
+
+		scope := getScope(req)
```

Comment:
In other middleware, we've just used the raw GetRequesrScope function and had a comment saying that it must always be initialised.

With this wrapper, checking it's not nil, we could end up in a situation where we are losing the data because the scope wasn't initialised properly. 

We should be consistent one way or another here IMO